crucible-llvm/src/Lang/Crucible/LLVM/MemModel/Generic.hs

```diff
@@ -1678,11 +1678,12 @@ asMemAllocationArrayStore sym w ptr mem
             MemWrite write_ptr write_source
               | Just write_blk_no <- asNat (llvmPointerBlock write_ptr)
               , blk_no == write_blk_no
-              , MemArrayStore arr (Just arr_store_sz_bv) <- write_source
-              , Just arr_store_sz <- BV.asUnsigned <$> asBV arr_store_sz_bv
-              , sz == arr_store_sz
-              , Just Refl <- testEquality w (ptrWidth write_ptr) ->
-                return $ Just arr
+              , MemArrayStore arr (Just arr_store_sz) <- write_source
+              , Just Refl <- testEquality w (ptrWidth write_ptr) -> do
+                sz_eq <- bvEq sym sz arr_store_sz
```

Comment:
Do we need to generalize this to a partial value?  Checking that the equality is concretely true is a bit limiting... I guess it's still sound if we just return `Nothing`, because instead we will fall back on the regular memory model operations, I think.

It would really help if you would add some description to the PR about why you're doing this generalization and what the expected results will be.

Reply:
it's sounds, but potentially (very) incomplete. There are going to be some incomplete interactions between the symbolic size allocations/writes and the concrete size allocations/writes, but that's unavoidable